storage/post_policy_v4.go

```diff
@@ -307,10 +318,18 @@ func GenerateSignedPostPolicyV4(bucket, object string, opts *PostPolicyV4Options
 	}
 
 	b64Policy := base64.StdEncoding.EncodeToString(condsAsJSON)
-	shaSum := sha256.Sum256([]byte(b64Policy))
-	signature, err := signingFn(shaSum[:])
-	if err != nil {
-		return nil, err
+	var signature []byte
+	var signErr error
+	// SignBytes is used only if SignRawBytes is not defined
+	if opts.SignRawBytes == nil && opts.SignBytes != nil {
```

Comment:
I worry about the fact that we are checking this condition multiple times. Maybe we just add an unexported field to the options about whether the signed bytes should be hashed or not, set that once when parsing the options, then make subsequent decisions accordingly?